modules/xfem/src/efa/EFAEdge.C

```diff
@@ -0,0 +1,419 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "EFAEdge.h"
+
+#include "EFANode.h"
+#include "EFAError.h"
+
+EFAEdge::EFAEdge(EFANode * node1, EFANode * node2):
+  _edge_node1(node1),
+  _edge_node2(node2)
+{
+  _embedded_nodes.clear();
+  _intersection_x.clear();
+  consistencyCheck();
+}
+
+EFAEdge::EFAEdge(const EFAEdge & other_edge)
+{
+  _edge_node1 = other_edge._edge_node1;
+  _edge_node2 = other_edge._edge_node2;
+  _intersection_x = other_edge._intersection_x;
+  _embedded_nodes = other_edge._embedded_nodes;
+  consistencyCheck();
+}
+
+EFAEdge::~EFAEdge() // do not delete edge node - they will be deleted
+{}                  // in EFAelement's destructor
+
+bool
+EFAEdge::equivalent(const EFAEdge & other) const
+{
+  bool isEqual = false;
+  if (other._edge_node1 == _edge_node1 &&
+      other._edge_node2 == _edge_node2)
+    isEqual = true;
+  else if (other._edge_node2 == _edge_node1 &&
+           other._edge_node1 == _edge_node2)
+    isEqual = true;
+  return isEqual;
+}
+
+bool
+EFAEdge::isPartialOverlap(const EFAEdge & other) const
+{
+  bool partially_overlap = false;
+  if (containsEdge(other) || other.containsEdge(*this))
+    partially_overlap = true;
+  return partially_overlap;
+}
+
+bool
+EFAEdge::containsEdge(const EFAEdge & other) const
+{
+  bool contains = false;
+  if (containsNode(other._edge_node1) &&
+      containsNode(other._edge_node2))
+    contains = true;
+  return contains;
+}
+
+bool
+EFAEdge::getNodeMasters(EFANode* node, std::vector<EFANode*> &master_nodes,
+                        std::vector<double> &master_weights) const
+{
+  master_nodes.clear();
+  master_weights.clear();
+  bool masters_found = false;
+  if (_edge_node1 == node)
+  {
+    master_nodes.push_back(_edge_node1);
+    master_weights.push_back(1.0);
+    masters_found = true;
+  }
+  else if (_edge_node2 == node)
```

Comment:
You might want to check if the first weight should really be `1.0` and not `0.0`.